src/app/core/common-components/customizable-tooltip/customizable-tooltip.directive.ts

```diff
@@ -0,0 +1,164 @@
+ * When a tooltip is only a string, the {@code MatTooltip} should be used instead.
+ *
+ * The Tooltip to render is provided as a template. This template is also the main argument
+ * to this directive. Place the directive on the HTML-Element where the tooltip should pop out
+ * when hovered over.
+ *
+ * @example
+ * <div [appCustomizableTooltip]="tooltip">Hover here to show the tooltip</div>
+ * <ng-template #tooltip>
+ *   Custom tooltip <i>with</i> HTML-Elements
+ * </ng-template>
+ *
+ * @see contentTemplate
+ */
+@Directive({
+  selector: "[appCustomizableTooltip]",
+})
+export class CustomizableTooltipDirective implements OnInit, OnDestroy {
+  /**
+   * Whether to disable the tooltip so it won't ever be shown
+   */
+  @Input() tooltipDisabled: boolean = false;
+
+  /**
+   * The amount of time in milliseconds that the user has to hover over the element before the tooltip
+   * is shown
+   */
+  @Input() delayShow: number = 1000;
+
+  /**
+   * The amount of time in milliseconds that the user's mouse has to leave the tooltip before it will
+   * be hidden
+   */
+  @Input() delayHide: number = 150;
+
+  /**
+   * The template that is shown in the tooltip.
+   * You can get the template ref of an HTML-Element by using the `#<template name>` syntax.
+   * An `<ng-template>` Element won't be shown to the user by default. Therefore it is the most commonly
+   * used element.
+   *
+   * @example
+   * <div [appCustomizableTooltip]="tooltip">Hover here to show the tooltip</div>
+   * <ng-template #tooltip>
+   *   Custom tooltip <i>with</i> HTML-Elements
+   * </ng-template>
+   */
+  @Input("appCustomizableTooltip") contentTemplate!: TemplateRef<any>;
+
+  /**
+   * Reference to the overlay (the Tooltip) to the control the visibility of the tooltip
+   * @private
+   */
+  private overlayRef!: OverlayRef;
+
+  /**
+   * The timeout used to deal both with the show-delay as well as the hide-delay
+   * @see delayHide
+   * @see delayShow
+   * @private
+   */
+  private timeoutRef?: ReturnType<typeof setTimeout>;
+
+  constructor(
+    private overlay: Overlay,
+    private overlayPositionBuilder: OverlayPositionBuilder,
+    private element: ElementRef
+  ) {}
+
+  ngOnInit(): void {
+    // Create a position strategy that determines where the overlay is positioned.
+    // In this case, it should be positioned at the bottom-center of the element.
+    const positionStrategy = this.overlayPositionBuilder
+      .flexibleConnectedTo(this.element)
+      .withPositions([
+        {
+          originX: "center",
+          originY: "bottom",
+          overlayX: "center",
+          overlayY: "top",
+          offsetY: 5,
+        },
+      ]);
+
+    this.overlayRef = this.overlay.create({ positionStrategy });
+  }
+
+  ngOnDestroy() {
+    this.hide();
+  }
+
+  @HostListener("mouseenter")
+  onMouseEnter() {
+    this.show();
+  }
+
+  @HostListener("mouseleave")
+  onMouseLeave() {
+    this.hide();
```

Comment:
There is a small problem with that. When you open the attendance-block tooltip at the bottom end of the screen, the tooltip is placed on top of the block component. This itself is not a problem but it results in this function being executed because the mouse leaves the block-component. This makes the tooltip flickering.

Any idea on how to solve this? It looks like we need to detect whether the mouse moved at all or make sure that the `mouseenter` event is executed on the tooltip.